Avoid asserts during constant folding binary ops on 'void' variables.

nishajain61 opened the pull request:
#### 8065ae84ebcd1350b24d83d8b473915ec5e8fb1c
<pre>
Avoid asserts during constant folding binary ops on &apos;void&apos; variables.
<a href="https://bugs.webkit.org/show_bug.cgi?id=271421">https://bugs.webkit.org/show_bug.cgi?id=271421</a>
<a href="https://rdar.apple.com/124085673">rdar://124085673</a>

Reviewed by NOBODY (OOPS!).

Fix asserts for example with / and % during constant folding when
the operands might be &apos;voids&apos;.

* Source/ThirdParty/ANGLE/src/compiler/translator/ParseContext.cpp:
(sh::TParseContext::binaryOpCommonCheck):
* Source/ThirdParty/ANGLE/src/tests/compiler_tests/Parse_test.cpp:
(TEST_F):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8065ae84ebcd1350b24d83d8b473915ec5e8fb1c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46244 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25376 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48831 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48936 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42285 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29728 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22833 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37793 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46822 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22470 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39879 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19031 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19850 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40979 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4288 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42554 "Found 1 new API test failure: TestWebKitAPI.DataDetectorTests.LoadWKWebViewWithDataDetectorTypePhoneNumber (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41333 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50726 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21245 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17745 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44976 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22540 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43890 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22907 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22235 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple reviewed (changes requested):
Need to get it landed via upstream. I provided the instructions.

nishajain61 committed:
Avoid asserts during constant folding binary ops on 'void' variables.
https://bugs.webkit.org/show_bug.cgi?id=271421
rdar://124085673

Reviewed by NOBODY (OOPS!).

Fix asserts for example with / and % during constant folding when
the operands might be 'voids'.

* Source/ThirdParty/ANGLE/src/compiler/translator/ParseContext.cpp:
(sh::TParseContext::binaryOpCommonCheck):
* Source/ThirdParty/ANGLE/src/tests/compiler_tests/Parse_test.cpp:
(TEST_F):
